### ISSUE TITLE
TLS support on Android 4.1

### ISSUE THREAD

**Issue by Rimco** [mendhak/gpslogger#637]
Related to #374: I have a custom URL configured which works fine with my own phone. However, I tried to setup the same URL on an older android phone and it failed. The SSL handshake fails in the logger, but is ok in the browser on the same phone.

The server is setup to accept all protocols except SSLv3. Even though, it seems that the logger is still trying to use an unsupported protocol (or cipher). Could it be related to the issue described in https://stackoverflow.com/questions/28943660/how-to-enable-tls-1-2-support-in-an-android-application-running-on-android-4-1?

Unfortunately it's not my phone so I can't debug the issue for prolonged periods of time.

**Comment by mendhak** [comment]
I've tried a lot but I haven't been able to get this to work. I've pushed up a few different commit attempts at this if anyone wants to peruse.

More notes

* Custom certificate validation is adding some complexity here, seems like bypassing validation would work but it's against the reason for doing this
* SNI is not working.  This may be the crucial bit, but OKHttp doesn't seem to be picking it up.  But it should be, unless something has changed in recent versions.

I considered, would it be possible to have TLS with only well known certs?  Implication - custom certs will fail.  SNI still won't work.  That's too drastic and too breaking a change.  

I'll leave this thread open for a while and close it later.  It remains unsolved.  I'll leave the branch around in case someone wants to try their hand at it.
